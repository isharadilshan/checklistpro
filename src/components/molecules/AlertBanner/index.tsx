import React from 'react';
import {Alert, HStack, Text, VStack} from 'native-base';

type AlertToastProps = {
  id?: string;
  status?: 'success' | 'error' | 'info' | 'warning';
  variant?: 'solid' | 'subtle' | 'left-accent' | 'top-accent' | 'outline';
  title: string;
  description: string;
};

const AlertToast: React.FC<AlertToastProps> = ({
  id,
  status,
  variant,
  title,
  description,
  ...rest
}) => {
  return (
    <Alert
      maxWidth="95%"
      alignSelf="center"
      flexDirection="row"
      status={status ? status : 'info'}
      variant={variant}
      {...rest}
    >
      <VStack space={1} flexShrink={1} w="100%">
        <HStack
          flexShrink={1}
          alignItems="center"
          justifyContent="space-between"
        >
          <HStack space={2} flexShrink={1} alignItems="center">
            <Alert.Icon />
            <Text
              fontSize="md"
              fontWeight="medium"
              flexShrink={1}
              color={
                variant === 'solid'
                  ? 'lightText'
                  : variant !== 'outline'
                  ? 'darkText'
                  : null
              }
            >
              {title}
            </Text>
          </HStack>
        </HStack>
        <Text
          px="6"
          color={
            variant === 'solid'
              ? 'lightText'
              : variant !== 'outline'
              ? 'darkText'
              : null
          }
        >
          {description}
        </Text>
      </VStack>
    </Alert>
  );
};

export default AlertToast;
