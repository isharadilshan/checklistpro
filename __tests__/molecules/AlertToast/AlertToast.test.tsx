import React from 'react';
import {describe, expect} from '@jest/globals';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import AlertBanner from '../../../src/components/molecules/AlertToast';

configure({adapter: new Adapter()});

const props = {
  name: 'TEST FORM SELECT CONTROLLER',
  defaultValue: 'test',
  error: 'This is not a valid form input',
  label: 'FORM SELECT',
  items: [
    {
      key: 1,
      label: 'ToDo',
      value: 'TODO',
    },
    {
      key: 2,
      label: 'In-Progress',
      value: 'INPROGRESS',
    },
    {
      key: 1,
      label: 'Hold',
      value: 'HOLD',
    },
    {
      key: 2,
      label: 'Done',
      value: 'DONE',
    },
  ],
};

describe('when rendering', () => {
  const wrapper = shallow(<AlertBanner {...props} />);

  it('should render a component matching the snapshot', () => {
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
    expect(wrapper).toHaveLength(1);
  });
});
