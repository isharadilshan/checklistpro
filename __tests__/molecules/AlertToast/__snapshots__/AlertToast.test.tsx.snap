// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when rendering should render a component matching the snapshot 1`] = `
<Memo(ForwardRef(Alert))
  alignSelf="center"
  defaultValue="test"
  error="This is not a valid form input"
  flexDirection="row"
  items={
    [
      {
        "key": 1,
        "label": "ToDo",
        "value": "TODO",
      },
      {
        "key": 2,
        "label": "In-Progress",
        "value": "INPROGRESS",
      },
      {
        "key": 1,
        "label": "Hold",
        "value": "HOLD",
      },
      {
        "key": 2,
        "label": "Done",
        "value": "DONE",
      },
    ]
  }
  label="FORM SELECT"
  maxWidth="95%"
  name="TEST FORM SELECT CONTROLLER"
  status="info"
>
  <Memo(ForwardRef(VStack))
    flexShrink={1}
    space={1}
    w="100%"
  >
    <Memo(ForwardRef(HStack))
      alignItems="center"
      flexShrink={1}
      justifyContent="space-between"
    >
      <Memo(ForwardRef(HStack))
        alignItems="center"
        flexShrink={1}
        space={2}
      >
        <Memo(ForwardRef(AlertIcon)) />
        <Memo(ForwardRef(Text))
          color="darkText"
          flexShrink={1}
          fontSize="md"
          fontWeight="medium"
        />
      </Memo(ForwardRef(HStack))>
    </Memo(ForwardRef(HStack))>
    <Memo(ForwardRef(Text))
      color="darkText"
      px="6"
    />
  </Memo(ForwardRef(VStack))>
</Memo(ForwardRef(Alert))>
`;
